### PR TITLE
Update Django to 4.2.15

### PR DESCRIPTION
**Pull request by jnm**
## Description

Incorporate Django security fixes; see https://www.djangoproject.com/weblog/2024/aug/06/security-releases/

### CODE REVIEW

File: dependencies/pip/dev_requirements.txt [hunk 1]
```diff
@@ -185,8 +176,6 @@ httmock==1.4.0
     # via -r dependencies/pip/dev_requirements.in
 idna==3.6
     # via requests
-importlib-metadata==7.0.2
-    # via markdown
```
Comment (jnm) [inline dependencies/pip/dev_requirements.txt:189]: `markdown` is still a dependency…not sure why this is changing

File: dependencies/pip/dev_requirements.txt [hunk 2]
```diff
@@ -32,15 +32,6 @@ azure-storage-blob==12.19.1
     # via django-storages
 backcall==0.2.0
     # via ipython
-backports-zoneinfo[tzdata]==0.2.1 ; python_version < "3.9"
```
Comment (jnm) [inline dependencies/pip/dev_requirements.txt:35]: I don't think this is a problem, but I'd appreciate your eyes on it, @noliveleger. See my related message on commit e4bc393a70a1041f3fb9dc8b244a7f540d6d95b5

File: dependencies/pip/dev_requirements.txt [hunk 3]
```diff
@@ -371,6 +357,4 @@ xlrd==2.0.1
     # via pyxform
 xlwt==1.3.0
     # via -r dependencies/pip/requirements.in
-zipp==3.18.0
-    # via importlib-metadata
```
Comment (jnm) [inline dependencies/pip/dev_requirements.txt:375]: another weird change related to `markdown` (dependency of a dependency)

**Comment by jnm** [comment]
GitLab CI failures seem unrelated to the changes
```
=========================== short test summary info ============================
FAILED onadata/apps/logger/tests/models/test_attachment.py::TestAttachment::test_create_thumbnails_command - django.core.exceptions.SuspiciousFileOperation: Detected path traversal attempt in '/tmp/test_media/bob/attachments/b9ebe5a3641c4cffaba708df38b5edea/5b2cc313-fc09-437e-8149-fcd32f695d41/1335783522563-large.jpg'
FAILED onadata/apps/logger/tests/models/test_attachment.py::TestAttachment::test_thumbnails - django.core.exceptions.SuspiciousFileOperation: Detected path traversal attempt in '/tmp/test_media/bob/attachments/ef6d15e45c00423088db7e06a10353f8/5b2cc313-fc09-437e-8149-fcd32f695d41/1335783522563-large.jpg'
====== 2 failed, 367 passed, 4 skipped, 326 warnings in 565.93s (0:09:25) ======
```

**Comment by noliveleger** [comment]
> GitLab CI failures seem unrelated to the changes
> 
> ```
> =========================== short test summary info ============================
> FAILED onadata/apps/logger/tests/models/test_attachment.py::TestAttachment::test_create_thumbnails_command - django.core.exceptions.SuspiciousFileOperation: Detected path traversal attempt in '/tmp/test_media/bob/attachments/b9ebe5a3641c4cffaba708df38b5edea/5b2cc313-fc09-437e-8149-fcd32f695d41/1335783522563-large.jpg'
> FAILED onadata/apps/logger/tests/models/test_attachment.py::TestAttachment::test_thumbnails - django.core.exceptions.SuspiciousFileOperation: Detected path traversal attempt in '/tmp/test_media/bob/attachments/ef6d15e45c00423088db7e06a10353f8/5b2cc313-fc09-437e-8149-fcd32f695d41/1335783522563-large.jpg'
> ====== 2 failed, 367 passed, 4 skipped, 326 warnings in 565.93s (0:09:25) ======
> ```

Seems to be related to `CVE-2024-39330` patched in `4.2.14`

File: dependencies/pip/dev_requirements.txt (hunk 3 above)
Comment (noliveleger) [reply dependencies/pip/dev_requirements.txt:375]: probably because Python 3.8 vs 3.10

File: dependencies/pip/dev_requirements.txt (hunk 2 above)
Comment (noliveleger) [reply dependencies/pip/dev_requirements.txt:35]: it is fine since - as you observed - it is readded by the script at the end of the file.

**Review by noliveleger — APPROVED** [review]

**Pull request merged**